### PR TITLE
Contributing to journal - rough guidelines and CI config

### PR DESCRIPTION
**Pull request by metazool**
* Outline of submission process based on the Transform discussion
  * Really rough travis config to discover and run notebooks, needs rethinking

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,13 +7,16 @@ install:
...
+  - pip install jupyter jupyter_client nbconvert nbval
 
 script:
   # Use Sphinx to make the html docs
   - cd docs
   - make html
   - touch _build/html/.nojekyll
-    
+  # roughtly execute every jupyter notebook we fine
+  - find . -name requirements.txt|xargs cat|xargs pip install -U
```
Comment (filippo82) [inline .travis.yml]: What do you want to achieve with this line?

File: .travis.yml [hunk 2]
```diff
@@ -7,13 +7,16 @@ install:
...
 
 script:
   # Use Sphinx to make the html docs
   - cd docs
   - make html
   - touch _build/html/.nojekyll
-    
+  # roughtly execute every jupyter notebook we fine
+  - find . -name requirements.txt|xargs cat|xargs pip install -U 
+  - find . -name "*.ipynb" |xargs jupyter nbconvert --to notebook --execute
```
Comment (filippo82) [inline .travis.yml]: This fails :/ Check the output of the Travis build.

Comment (metazool) [reply .travis.yml]: I'd not expected it to be run in a scenario where there are no notebooks to find. In the short term, installing requirements for and re-testing all journal submissions isn't a problem

Comment (metazool) [reply .travis.yml]: Also thank you for comments! I'm replacing this hacky pipeline with a slightly less hacky pipeline now

File: .travis.yml (hunk 1 above)
Comment (metazool) [reply .travis.yml]: I've replaced this now, see comment below

**Comment by metazool** [comment]
Hi @filippo82 can i trouble you to cast your eyes over this again?

  * moved the travis script lines that are installing the requirements and trying to validate the notebook, into a shell script
  * pipeline is still failing, but at a later stage during the notebook execution, over a genuine python SyntaxError related to import styles, which @johnjarmitage is addressing

**Comment by metazool** [comment]
Closing this for now and moving the thrashing about with dependencies to a fork of the repo

**Pull request closed without merging**